gschem/src/x_newtext.c

```diff
@@ -292,7 +292,11 @@ static void newtext_init(NewText *dialog)
                     0);
 
   dialog->textsizecb = gschem_integer_combo_box_new();
-  gschem_integer_combo_box_set_value(dialog->textsizecb, 12);
+
+  GschemToplevel *w_current = g_current_window ();
+  gschem_integer_combo_box_set_value (dialog->textsizecb,
+    w_current ? w_current->text_size : default_text_size);
```

Comment:
First somebody should go into the code and clearly understand what it does, I think ;-)